src/main/java/uk/ac/ebi/ait/filecontentvalidatorservice/service/FileContentValidationHandler.java

```diff
@@ -27,13 +31,15 @@
     ReportFileConfig reportFileConfig;
     ReadsValidator validator;
     ReadsValidationResponse validationResponse;
+    CommandLineParameters commandLineParameters;
 
     private File validationDir;
     private File processDir;
```

Comment:
validationResponse, validationDir and processDir are sensitive to multithreading synchronization problems.

please use the keyword final with other members that are being initialized in the contructor.

Reply:
It is a command line app executed by the file processing listener app. I might be wrong, but multithreading is not really applied here.